File: common/src/main/java/dev/mayaqq/estrogen/registry/common/EstrogenEffects.java
package dev.mayaqq.estrogen.registry.common;

import dev.architectury.registry.registries.Registrar;
import dev.mayaqq.estrogen.registry.common.effects.EstrogenEffect;
import net.minecraft.core.Registry;
import net.minecraft.world.effect.MobEffect;
import net.minecraft.world.effect.MobEffectCategory;

import static dev.mayaqq.estrogen.Estrogen.MANAGER;
import static dev.mayaqq.estrogen.Estrogen.id;

public class EstrogenEffects {

    public static final Registrar<MobEffect> MOB_EFFECTS = MANAGER.get().get(Registry.MOB_EFFECT);

    public static final MobEffect ESTROGEN_EFFECT = new EstrogenEffect(MobEffectCategory.BENEFICIAL, 104164161);

    public static void register() {
        MOB_EFFECTS.register(id("estrogen"), () -> ESTROGEN_EFFECT);
    }
}

File: src/main/kotlin/dev/mayaqq/estrogen/content/EstrogenEffects.kt
package dev.mayaqq.estrogen.content

import dev.mayaqq.estrogen.MOD_ID
import dev.mayaqq.estrogen.content.effects.DreamingEffect
import dev.mayaqq.estrogen.content.effects.EstrogenEffect
import dev.mayaqq.estrogen.utils.EstrogenColors
import invoke.kitty.kritter.registry.api.Registrar
import invoke.kitty.kritter.registry.api.builder.entry
import invoke.kitty.kritter.registry.api.entry.RegistryEntry
import net.minecraft.core.registries.Registries
import net.minecraft.world.effect.MobEffect
import net.minecraft.world.effect.MobEffectCategory

@Suppress("unused")
object EstrogenEffects : Registrar<MobEffect> by Registrar(MOD_ID, Registries.MOB_EFFECT) {
    @JvmStatic
    val Estrogen: RegistryEntry<MobEffect> = entry("estrogen", {EstrogenEffect(MobEffectCategory.BENEFICIAL, 104164161)}) {}

    val Dreaming: RegistryEntry<MobEffect> = entry("dreaming", fun() = DreamingEffect(MobEffectCategory.NEUTRAL, EstrogenColors.DREAM_BLOCK)) {}
}